### PR TITLE
chore(tests): Test with `default` feature in CI

### LINKED ISSUE #2673: Not all tests are running in CI
Tests for at least some of the `sources` aren't being run in CI. For example for `socket` source tests in `/src/sources/socket/mod.rs` aren't being run.

Noting, all those tests are being run locally and they are passing.

An additional CI check for `sources` feature could be added.

### PR DESCRIPTION
**Pull request by ktff**
Closes #2673

Most of `source`, `transform`, and `sink` tests haven't been running in CI or more accurately they aren't tested with `USE_CONTAINER=none`, but are with a container.   

This PR: 
 * Adds `test-default` which runs tests in library with `default` feature. 
 * Changes `aws-ec2-metadata-integration-tests` feature with  `aws-integration-tests`, since it wasn't being used anywhere else so it wasn't run at all.
 * Updates `syslog` tests that were failing.

### Open questions

- [x] We could split testing for `sources`, `transforms`, and `sinks`. But either we would need to compile the same things three times, or we could compile with their features but anytime that we add a test that uses a component from other group we would need to remember to update the feature list with the additional feature. Currently only some source tests use sinks. (EDIT: No. There are changes coming to how we test, so let's keep it simple)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -598,6 +598,24 @@ services:
...
+      RUST_BACKTRACE: full
+    volumes:
+      - $PWD:$PWD
+      - ./target/x86_64-unknown-linux-musl/cargo/registry:/opt/rust/cargo/registry
+      - ./target/x86_64-unknown-linux-musl/cargo/git:/opt/rust/cargo/git
+      - ./target/x86_64-unknown-linux-musl/rustup/tmp:/opt/rust/rustup/tmp
+    network_mode: host
+    working_dir: $PWD
+    user: $USER
+    command: cargo test --lib --no-default-features --features default-cmake --target x86_64-unknown-linux-musl
```
Comment (Hoverbear) [inline docker-compose.yml:617]: This isn't exactly `default` is it?

**Comment by Hoverbear** [comment]
This seems to test with the `default-cmake` feature not the `default` feature.

**Comment by Hoverbear** [comment]
This seems to be a duplicate of `test-unit`?

File: docker-compose.yml (hunk 1 above)
Comment (ktff) [reply docker-compose.yml:617]: It's mostly equivalent to `default` and I didn't want to use it here since `docker-compose` uses `default-cmake`. Although I'm guessing that `default` wouldn't work here. But it used in the script which gets called in CI.

**Comment by ktff** [comment]
`test-unit` tests without any feature. 

And the script that is being called in CI is using `default` feature.

**Comment by ktff** [comment]
As per we agreed in slack @Hoverbear , I'll merge this.
